### PR TITLE
New Resource: Waypoint Add-on

### PR DESCRIPTION
**Pull request by HenryEstberg**
### :hammer_and_wrench: This adds a new resource `hcp_waypoint_add_on`.

Example config:
```
resource "hcp_waypoint_add_on" "test" {
  name    = "Some_Test_Add-On"
  application = {
    id = hcp_waypoint_application_template.test.id
  }
  definition = {
    name = hcp_waypoint_add_on_definition.test.name
  }
}
```

Output from acceptance testing:

```sh
$ make testacc TESTARGS='-run=TestAccWaypoint_Add_On_basic'

...
```

### CODE REVIEW

**Review by catsby — COMMENTED** [review]
Question about the 500 error

File: docs/data-sources/waypoint_add_on.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "hcp_waypoint_add_on Data Source - terraform-provider-hcp"
+subcategory: ""
```
[the remark this answers is not in the document]
Comment (catsby) [reply docs/data-sources/waypoint_add_on.md:4]: https://github.com/hashicorp/terraform-provider-hcp/pull/809

File: internal/provider/waypoint/resource_waypoint_add_on_test.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+		loc := &sharedmodels.HashicorpCloudLocationLocation{
+			OrganizationID: orgID,
+			ProjectID:      projectID,
+		}
+
+		addOn, err := clients.GetAddOnByID(context.Background(), client, loc, id)
+		if err != nil {
+			// expected (500 because the application is destroyed as well)
+			if clients.IsResponseCodeInternalError(err) {
```
[the remark this answers is not in the document]
Comment (briancain) [reply internal/provider/waypoint/resource_waypoint_add_on_test.go:102]: I bet this is an add-on bug, 500 is (often incorrectly) returned at the service layer if our `hcerr.Externalize` doesn't know what protobuf status code it should have.

File: docs/data-sources/waypoint_add_on.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (HenryEstberg) [reply docs/data-sources/waypoint_add_on.md:4]: Huh interesting, thank you for looking into that!

File: internal/provider/waypoint/resource_waypoint_add_on_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (HenryEstberg) [reply internal/provider/waypoint/resource_waypoint_add_on_test.go:105]: The fix is in this [Waypoint PR](https://github.com/hashicorp/cloud-waypoint-service/pull/1109/files#diff-dd946e7bd785e234ee6592d1b74a80312f3a1abcb3edc9435780f10793bf8595R395). Once that gets merged, I will update this PR with the correct 404 error code.

**Review by catsby — APPROVED** [review]

File: internal/provider/waypoint/resource_waypoint_add_on_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (catsby) [reply internal/provider/waypoint/resource_waypoint_add_on_test.go]: That PR was merged, resolving this

**Comment by catsby** [comment]
@manish-hashicorp this is good-to-go when you have a moment, thank you

**Comment by catsby** [comment]
Hey @hashicorp/cloud-foundations could someone please take a look here? We'd like to merge this and move forward and hopefully get this released tomorrow

**Review by manish-hashicorp — APPROVED** [review]
Rubber stamping.

**Comment by catsby** [comment]
@manish-hashicorp thank you! @HenryEstberg please merge when you can

**Pull request merged**